src/main/java/org/folio/circulation/infrastructure/storage/inventory/InstanceRepository.java

```diff
@@ -55,37 +54,27 @@ public InstanceRepository(Clients clients) {
       return completedFuture(succeeded(multipleRequests));
     }
 
-    final FindWithMultipleCqlIndexValues<Instance> fetcher = createInstancesFetcher();
-
-    return fetcher.findByIds(instanceIdsToFetch)
+    return  findWithMultipleCqlIndexValues(instancesClient, "instances", Instance::from)
+      .findByIds(instanceIdsToFetch)
       .thenApply(multipleInstancesResult -> multipleInstancesResult.next(
         multipleInstances -> {
-          List<Request> newRequestList = new ArrayList<>();
-          Collection<Instance> instanceCollection = multipleInstances.getRecords();
-          for (Request request : requests) {
-            Request newRequest = null;
-            boolean foundInstance = false;
-            for (Instance instance : instanceCollection) {
-              if (request.getInstanceId() != null && request.getInstanceId().equals(instance.getId())) {
-                newRequest = request.withInstance(instance);
-                foundInstance = true;
-                break;
+          List<Request> newRequestList = requests.stream()
+            .map(request -> {
```

Comment:
Can we move this mapping function to a separate method?

Reply:
done